### PR TITLE
[MRG] MAINT: read_spikes -> cell_response.py

### PR DESCRIPTION
**Pull request by jasmainak**
Cruft from #305

### CODE REVIEW

File: hnn_core/cell_response.py [hunk 1]
```diff
@@ -393,3 +394,58 @@ def write(self, fname):
...
+    spike_gids = list()
+    spike_types = list()
+    for file in sorted(glob(str(fname))):
+        spike_trial = np.loadtxt(file, dtype=str)
+        if spike_trial.shape[0] > 0:
+            spike_times += [list(spike_trial[:, 0].astype(float))]
+            spike_gids += [list(spike_trial[:, 1].astype(int))]
+
+            # Note that legacy HNN 'spk.txt' files don't contain a 3rd column
+            # for spike type. If reading a legacy version, validate that a
```
Comment (ntolley) [inline hnn_core/cell_response.py:429]:
```suggestion
            # for spike type. If reading a legacy version, ensure that a
```
Just kind of awkward sounding reading this again.

File: hnn_core/cell_response.py [hunk 2]
```diff
@@ -393,3 +394,58 @@ def write(self, fname):
...
+            if spike_trial.shape[1] == 3:
+                spike_types += [list(spike_trial[:, 2].astype(str))]
+            else:
+                if gid_ranges is None:
+                    raise ValueError("gid_ranges must be provided if spike "
+                                     "types are unspecified in the "
+                                     "file %s" % (file,))
+                spike_types += [[]]
+        else:
+            spike_times += [[]]
```
Comment (ntolley) [inline hnn_core/cell_response.py:440]:
```suggestion
            spike_times += [list()]
```

File: hnn_core/cell_response.py [hunk 3]
```diff
@@ -393,3 +394,58 @@ def write(self, fname):
...
+            spike_gids += [[]]
```
Comment (ntolley) [inline hnn_core/cell_response.py:441]:
```suggestion
            spike_gids += [list()]
```

File: hnn_core/cell_response.py [hunk 4]
```diff
@@ -393,3 +394,58 @@ def write(self, fname):
...
+            spike_types += [[]]
```
Comment (ntolley) [inline hnn_core/cell_response.py:442]:
```suggestion
            spike_types += [list()]
```

File: hnn_core/cell_response.py [hunk 5]
```diff
@@ -393,3 +394,58 @@ def write(self, fname):
...
+            # for spike type. If reading a legacy version, validate that a
+            # gid_dict is provided.
+            if spike_trial.shape[1] == 3:
+                spike_types += [list(spike_trial[:, 2].astype(str))]
+            else:
+                if gid_ranges is None:
+                    raise ValueError("gid_ranges must be provided if spike "
+                                     "types are unspecified in the "
+                                     "file %s" % (file,))
+                spike_types += [[]]
```
Comment (ntolley) [inline hnn_core/cell_response.py:438]:
```suggestion
                spike_types += [list()]
```

Comment (ntolley) [reply hnn_core/cell_response.py:438]: possibly more clear? the constructor doesn't work for `list(list())` or `list([])` interestingly

**Review by ntolley — APPROVED** [review]

**Comment by ntolley** [comment]
Just realized, are there no tests for the `read_spikes` function?

**Comment by jasmainak** [comment]
there are tests actually, see here for example: https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_cell_response.py#L28-L29. 

feel free to merge if/when you're happy. It's a straightforward PR just moving things around.

**Comment by ntolley** [comment]
Ah I see, I think my confusion was why none of the test files had to modify the imports (but read_spikes is actually in the __init__). Embarrassingly I think I may have written some of those tests 😄

**Pull request merged**